test_scripts/Defects/4_6/964_SDL_sends_to_mobile_APPLICATION_NOT_REGISTRED_in_setAppIcon_responce_if_HMI_responds_with_INVALID_DATA.lua

```diff
@@ -0,0 +1,74 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/SmartDeviceLink/sdl_core/issues/964
+--
+-- Precondition:
+-- SDL Core and HMI are started. App is registered, HMI level = FULL
+-- Description:
+-- Steps to reproduce:
+-- 1) Mobile app sends PutFile(request).
+-- 2) Mobile app sends invalid SetAppIcon(requests).
+-- Expected:
+-- 1) Respond successfully processed PutFile requests.
+-- 2) Send to mobile resultCode(INVALID_DATA) which was sent to SDL by HMI.
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Defects/commonDefects')
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+local utils = require("user_modules/utils")
+
+--[[ Local Variables ]]
+local function getConfigAppParams(pAppId)
+  if not pAppId then pAppId = 1 end
+  return config["application" .. pAppId].registerAppInterfaceParams
+end
+
+local function getPathToFileInStorage(pFileName, pAppId)
+  if not pAppId then pAppId = 1 end
+  return commonPreconditions:GetPathToSDL() .. "storage/"
+    .. getConfigAppParams(pAppId).appID .. "_"
+    .. utils.getDeviceMAC() .. "/" .. pFileName
+end
+
+local function putFile(self)
+  local cid = self.mobileSession1:SendRPC( "PutFile",
+    {syncFileName = "\\syncFileName", fileType = "GRAPHIC_PNG", persistentFile = false, systemFile = false},
+  "files/\\syncFileName")
+
+  self.mobileSession1:ExpectResponse(cid, { success = true, resultCode = "SUCCESS"})
+end
+
+local requestParams = { syncFileName = "\\syncFileName" } -- invalid type of parameter
+
+local requestUiParams = {
+  syncFileName = {
+    imageType = "DYNAMIC",
+    value = getPathToFileInStorage(requestParams.syncFileName)
+  }
+}
+
+local allParams = {requestParams = requestParams, requestUiParams = requestUiParams }
```

Comment:
Please move all variables in section --[[ Local Variables ]]

and make --[[ Local Functions ]] section and write functions in it